### PR TITLE
[Merged by Bors] - Cleanup code and changelog

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

This updates the changelog after the release of `v1.6.6-hotfix1` and fixes a few typos in code.

## Description

- Updates CHANGELOG
- all code changes are typo fixes except
- `activation/handler_v2.go`: simplified code to not having to query an identity again about its standing
  - before if an identity was found to be malfeasant `checkMalfeasance` would return and then we would query again if it was found to be malicious, now `checkMalfeasance` just returns `true` if an identity is considered a bad actor.

## Test Plan

- existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: hare3/hare.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ type Opt func(*Hare)
 
 func WithWallclock(clock clockwork.Clock) Opt {
 	return func(hr *Hare) {
-		hr.wallclock = clock
+		hr.wallClock = clock
 	}
 }
```
Comment (poszu) [inline hare3/hare.go]: Should the function be called `WithWallClock` then?

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**